setup/index.md

```diff
@@ -4,22 +4,45 @@ title: Testing Setup
 root: ..
 ---
 
-This directory contains scripts for testing your machine to make sure
+This directory contains scripts for installing the 
+environment needed for the course and testing your machine to make sure
 you have the software you'll need for your workshop installed.  To use
 these scripts:
 
-1.  Download [swc-installation-test-1.py](swc-installation-test-1.py).
+1.  If you are using Linux or Mac download [`swc-installation.sh`](swc-installation.sh), if you are
+    using Windows, download [`swc-installation.ps1`](swc-installation.ps1) instead.
 
-2.  Run it from the shell:
+    Also download the [`requirements.txt`](requirements.txt) file. If you are using a new apple
+    mac M1 or M2, then download [`requirements_apple_m.txt`](requirements_apple_m.txt) too.
+
+2.  If you are running **Windows**, then you can click with the right button on the
+    `swc-installation.ps1` that you downloaded and select `Run with PowerShell`. This may prompt a
+    question about Change of policies, if so answer <key>Y</key>es.
```

Comment:
@davecash75 @tdveale this runs on my windows VM. Would you be able to test it? If not, I'm happy for this to be merged as it is.

Reply:
Thanks @dpshelio ! The powershell setup runs well on my windows 10 machine.

However I hit an error (windows and Mac) when running `python swc-installation-test-2.py` @davecash75 :

```
Traceback (most recent call last):
  File "/Users/tom/git/2023-03-09-UCL-DasH/setup/swc-installation-test-2.py", line 64, in <module>
    import distro as _dist
ModuleNotFoundError: No module named 'distro'
```

Reply:
Hi Tom, perhaps distro is not in the windows list of available packages? Could you try a pip install distro within that environment and see what happens?

Reply:
Sorry this issue on Mac was my mistake.

Running more tests on Windows 10 now - will update soon.

Reply:
### On Mac (intel)
- Installation and set up scripts run fine
- Able to run jupyter lab
- regex imports now 

### On Windows 10
- The `bash swc-installation.sh` and both test scripts all run through fine if done in git bash (one minor warning about cuda but ultimately passed).
- I ran into issues if I ran `swc-installation.ps1` in powershell and then `python swc-installation-test-2.py` in git bash (lots of fails in second test script). First test script runs through fine. See output below:

```
$ python swc-installation-test-2.py
check command line shell (virtual-shell)...     pass
check text/code editor (virtual-editor)...      pass
check web browser (virtual-browser)...  pass
check Git (git)...      pass
check PyPI installer (virtual-pypi-installer)...        pass
check Setuptools (setuptools)...        pass
check SQLite 3 (sqlite3)...     pass
check SQLite Python package (sqlite3-python)... pass
check Python version (python)...        pass
check IPython script (ipython)...       pass
check Jupyter (jupyter)...      fail
check Jupyter Lab (jupyterlab)...       fail
check Argparse (argparse)...    pass
check NumPy (numpy)...  fail
check SciPy (scipy)...  fail
check Matplotlib (matplotlib)...        fail
check Nibabel (nibabel)...      fail
check Nilearn (nilearn)...      fail
check Open CV (cv2)...  fail
check Pandas (pandas)...        fail
check Pillow (PIL)...   fail
check Seaborn (seaborn)...      fail
check scikit-image (skimage)... fail
check Tensorflow (tensorflow)...        fail

Successes:

command line shell (virtual-shell) Bourne Again Shell (bash) 5.1.16
text/code editor (virtual-editor) Vim (vim) 8.2
web browser (virtual-browser) Firefox (firefox) 110.0.1
Git (git) 2.37.2
PyPI installer (virtual-pypi-installer) pip 23.0.1
Setuptools (setuptools) 65.6.3
SQLite 3 (sqlite3) 3.40.1
SQLite Python package (sqlite3-python) 3.9.16 (main, Mar  1 2023, 18:30:21) [MSC v.1916 64 bit (AMD64)]
Python version (python) 3.9.16 (main, Mar  1 2023, 18:30:21) [MSC v.1916 64 bit (AMD64)]
IPython script (ipython) 8.2.0
Argparse (argparse) 1.1

Failures:

check for Jupyter (jupyter) failed:
  some dependencies for Jupyter (jupyter) were not satisfied
  For instructions on installing an up-to-date version, see
  http://jupyter.org/install.html
  causes:
  check for Jinja (jinja) failed:
    could not import the 'jinja2' package for Jinja (jinja)
    For instructions on installing an up-to-date version, see
    http://jinja.pocoo.org/docs/intro/#installation

check for Jupyter Lab (jupyterlab) failed:
  could not import the 'jupyterlab' package for Jupyter Lab (jupyterlab)
  For instructions on installing an up-to-date version, see
  https://jupyter.org/install

check for NumPy (numpy) failed:
  could not import the 'numpy' package for NumPy (numpy)
  For instructions on installing an up-to-date version, see
  http://docs.scipy.org/doc/numpy/user/install.html

check for SciPy (scipy) failed:
  could not import the 'scipy' package for SciPy (scipy)
  For instructions on installing an up-to-date version, see
  http://www.scipy.org/Installing_SciPy/Windows

check for Matplotlib (matplotlib) failed:
  could not import the 'matplotlib' package for Matplotlib (matplotlib)
  For instructions on installing an up-to-date version, see
  http://matplotlib.org/users/installing.html#installing-on-windows

check for Nibabel (nibabel) failed:
  could not import the 'nibabel' package for Nibabel (nibabel)
  For instructions on installing an up-to-date version, see
  http://software-carpentry.org/setup/

check for Nilearn (nilearn) failed:
  could not import the 'nilearn' package for Nilearn (nilearn)
  For instructions on installing an up-to-date version, see
  http://software-carpentry.org/setup/

check for Open CV (cv2) failed:
  could not import the 'cv2' package for Open CV (cv2)
  For instructions on installing an up-to-date version, see
  http://software-carpentry.org/setup/

check for Pandas (pandas) failed:
  could not import the 'pandas' package for Pandas (pandas)
  For instructions on installing an up-to-date version, see
  http://pandas.pydata.org/pandas-docs/stable/install.html

check for Pillow (PIL) failed:
  could not import the 'PIL' package for Pillow (PIL)
  For instructions on installing an up-to-date version, see
  http://software-carpentry.org/setup/

check for Seaborn (seaborn) failed:
  could not import the 'seaborn' package for Seaborn (seaborn)
  For instructions on installing an up-to-date version, see
  http://software-carpentry.org/setup/

check for scikit-image (skimage) failed:
  could not import the 'skimage' package for scikit-image (skimage)
  For instructions on installing an up-to-date version, see
  http://software-carpentry.org/setup/

check for Tensorflow (tensorflow) failed:
  could not import the 'tensorflow' package for Tensorflow (tensorflow)
  For instructions on installing an up-to-date version, see
  http://software-carpentry.org/setup/
(python_course)
```

Reply:
Tom just to check did you activate the environment before running the installation test?

Reply:
Yep it's activated

Reply:
Ah so the issue is if you run it through powershell (a Windows process) but then try to activate and test in git bash? I think that this could be git bash is inherently a small GNU environment inside of Windows and so it thinks it is Linux while the powershell has installed thinking it is Windows? @dpshelio  any ideas. Should tests and other things be run through powershell or cmd instead of git bash if that was how it was installed?

Reply:
it may be that the powershell script is not actually installing the things? Did you execute via clicking or on the terminal?

Reply:
I've been right clicking and running via powershell. Although I just got a security warning when trying via the powershell terminal. I don't get a security warning pop-up like it suggests in the instructions. 

The powershell does close when running the `swc-installation.ps1` but I thought this was a feature of the process finishing - maybe it crashes/exits due to permissions?

Reply:
I've tested all now, with the python script too (and installing only a subset of the packages, e.g., jupyter) and the test works fine. The shell check produces many errors if one is not detected. I've removed in my next commit.